google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -192,19 +196,54 @@ private static String getAppEngineInstanceName() {
     return System.getenv("GAE_INSTANCE");
   }
 
-  private static List<LoggingEnhancer> getEnhancers(Resource resourceType) {
-    List<LoggingEnhancer> enhancers;
+  private static List<LoggingEnhancer> createEnhancers(Resource resourceType) {
+    List<LoggingEnhancer> enhancers = new ArrayList<>(2);
     switch (resourceType) {
       // Trace logging enhancer is supported on GAE Flex and Standard.
-      case GaeAppStandard:
       case GaeAppFlex:
-        enhancers = new ArrayList<>();
-        enhancers.add(new TraceLoggingEnhancer());
+        enhancers.add(new LabelLoggingEnhancer(
+            APPENGINE_LABEL_PREFIX, Collections.singletonList(Label.InstanceName)));
+        enhancers.add(new TraceLoggingEnhancer(APPENGINE_LABEL_PREFIX));
+        break;
+      case GaeAppStandard:
+        enhancers.add(new TraceLoggingEnhancer(APPENGINE_LABEL_PREFIX));
         break;
       default:
-        enhancers = Collections.emptyList();
         break;
     }
     return enhancers;
   }
+
+  /**
+   * Adds additional resource-based labels to log entries.
+   * Labels that can be provided with {@link MonitoredResource.Builder#addLabel(String, String)}
+   * are restricted to a supported set per resource.
+   *
+   * @see <a href="https://cloud.google.com/logging/docs/api/v2/resource-list">Logging Labels</a>
+   */
+  private static class LabelLoggingEnhancer implements LoggingEnhancer {
+
+    Map<String, String> labels;
```

Comment:
Please keep class fields private, unless it is absolutely necessary to do otherwise.